drf_kit/managers/availability_managers.py

```diff
@@ -0,0 +1,117 @@
+from dataclasses import dataclass
+from datetime import datetime
+
+from django.db import models
+from django.db.models import Q
+from django.utils import timezone
+
+
+@dataclass
+class AvailabilityChecker:
```

Comment:
Should we ensure those attributes are `datetime` type?

Reply:
We always can, but I don't think it's really necessary. Some flexibility for other date types is supported, and `mypy` can enforce the typing.

I'm willing to start using it so we can see how bad use cases we can come up with, then we enforce a more strict usage.